fl4health/clients/fenda_client.py

```diff
@@ -118,24 +147,13 @@ def get_cosine_similarity_loss(self, local_features: torch.Tensor, global_featur
         assert len(local_features) == len(global_features)
         return torch.abs(self.cos_sim(local_features, global_features)).mean()
 
-    def compute_contrastive_loss(
-        self, features: torch.Tensor, positive_pairs: torch.Tensor, negative_pairs: torch.Tensor
-    ) -> torch.Tensor:
+    def get_mkmmd_loss(self, local_features: torch.Tensor, global_features: torch.Tensor) -> torch.Tensor:
         """
-        Contrastive loss aims to enhance the similarity between the features and their positive pairs
-        while reducing the similarity between the features and their negative pairs.
+        MK-MMD loss aims to compute the distance among distribtution of current local features
+        and current global features of fenda model.
         """
-        assert self.temperature is not None
-        assert len(features) == len(positive_pairs)
-        posi = self.cos_sim(features, positive_pairs)
-        logits = posi.reshape(-1, 1)
-        assert len(features) == len(negative_pairs)
-        nega = self.cos_sim(features, negative_pairs)
-        logits = torch.cat((logits, nega.reshape(-1, 1)), dim=1)
-        logits /= self.temperature
-        labels = torch.zeros(features.size(0)).to(self.device).long()
-
-        return self.ce_criterion(logits, labels)
+        assert len(local_features) == len(global_features)
```

Comment:
We probably want to assert that the shape of these features is equal, not just the batch size?

Reply:
That is better assertion